src/irmin-pack/inode.ml

```diff
@@ -1776,7 +1776,10 @@ struct
     type key = Key.t
     type t = T.key Bin.t [@@deriving irmin]
     type metadata = T.metadata [@@deriving irmin]
+    type Pack_value.kinded += Node of t
 
+    let to_kinded t = Node t
+    let of_kinded = function Node n -> n | _ -> assert false
```

Comment:
I think I would prefer if this extensible type was hidden in the LRU implementation.. I think it should be possible to create behind the scene the fresh `+= Node` new type constructor by keeping the LRU a functor to instantiate for each type of value that we want to store in it.

Reply:
Interesting idea! I'll see what I can do. I do agree it would be nicer to hide this bookkeeping.